### PR TITLE
Prep for v1.15.0

### PR DESCRIPTION
**Pull request by odow**
Changelog preview: https://jump.dev/JuMP.jl/previews/PR3485/release_notes/

## TODO

 - [x] Add a note for https://github.com/jump-dev/JuMP.jl/pull/3474#issuecomment-1704504740
 - [x] https://github.com/jump-dev/JuMP.jl/pull/3483
 - [x] #3488 
 - [x] #3490 
 - [x] #3491
 - [x] #3492 
 - [x] #3489 
 - [x] #3494
 - [x] #3495
 - [x] #3497
 - [x] #3500
 - [x] #3503
 - [x] #3504
 - [ ] #3506 
 - [x] #3507
 - [x] #3509

## Pre-release

 - [x] Check that the pinned packages in `docs/Project.toml` are updated. We pin
       the versions so that changes in the solvers (changes in printing, small
       numeric changes) do not break the printing of the JuMP docs in arbitrary
       commits.
 - [x] Check that the `rev` fields in `docs/packages.toml` are updated. We pin
       the versions of solvers and extensions to ensure that changes to their
       READMEs do not break the JuMP docs in arbitrary commits, and to ensure
       that the versions are compatible with the latest JuMP and
       MathOptInterface releases.
 - [x] Update `docs/src/changelog.md`
 - [x] https://github.com/jump-dev/JuMP.jl/actions/runs/6178384899
 - [x] Change the version number in `Project.toml`
 - [x] Update the links in README.md
 - [x] The commit messages in this PR do not contain `[ci skip]`

## The release

 - [ ] After merging this pull request, comment `[at]JuliaRegistrator register` in
       the GitHub commit. This should automatically publish a new version to the
       Julia registry, as well as create a tag, and rebuild the documentation
       for this tag.

       These steps can take quite a bit of time (1 hour or more), so don't be
       surprised if the new documentation takes a while to appear. In addition,
       the links in the README will be broken until JuliaHub fetches the new
       version on their servers.

## Post-release

 - [ ] Once the tag is created, update the relevant `release-` branch. The latest
       release branch at the time of writing is `release-1.0` (we haven't
       back-ported any patches that needed to create a `release-1.Y` branch). To
       to update the release branch with the v1.10.0 tag, do:
       ```
       git checkout release-1.0
       git pull
       git merge v1.10.0
       git push
       ```

### CODE REVIEW

File: docs/src/changelog.md [hunk 1]
```diff
@@ -7,6 +7,51 @@ CurrentModule = JuMP
...
+
+ - The printed form of the expression may change, for example from `x * y` to
+   `y * x`. This can cause tests which test the `String` representation of a
+   model to fail.
+ - Because of the different order of operations, some coefficients may change
+   due to floating point round-off error.
+ - Because of the different order of operations, and particularly when working
+   with a JuMP extension, you may encounter a `MethodError` due to a missing or
+   ambiguous method. These errors are due to previously existing bugs that were
+   not triggered by the previous parsing code. If you encouter such an error,
```
Comment (odow) [inline docs/src/changelog.md:32]:
```suggestion
   not triggered by the previous parsing code. If you encounter such an error,
```

**Comment by odow** [comment]
@blegat, do we want to fix https://github.com/jump-dev/MutableArithmetics.jl/pull/225 first?

File: docs/src/changelog.md [hunk 2]
```diff
@@ -7,6 +7,75 @@ CurrentModule = JuMP
...
+   type-stable `x * x` instead of `x^2`.
+
+### Added
+
+ - Added [`triangle_vec`](@ref) which simplifies adding [`MOI.LogDetConeTriangle`](@ref)
+   and [`MOI.RootDetConeTriangle`](@ref) constraints (#3456)
+ - Added the new nonlinear interface. This is a very large change. See the
+   documentation at [Nonlinear Modeling](@ref) and the (long) discussion in
+   [JuMP.jl#3106](https://github.com/jump-dev/JuMP.jl/pull/3106). Related PRs
+   are (#3468) (#3472) (#3475) (#3483) (#3487)
```
Comment (odow) [inline docs/src/changelog.md:61]:
```suggestion
   are (#3468) (#3472) (#3475) (#3483) (#3487) (#3488)
```

**Comment by blegat** [comment]
I just release a new version of MA so we can go ahead and release JuMP v1.15, can't wait :tada:

File: docs/src/changelog.md [hunk 3]
```diff
@@ -7,6 +7,75 @@ CurrentModule = JuMP
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## Version 1.15.0 (unreleased)
```
Comment (odow) [inline docs/src/changelog.md:10]: TODO

**Comment by pulsipher** [comment]
> https://github.com/jump-dev/JuMP.jl/actions/runs/6094919151

The failure with InfiniteOpt is minor and has to do with expression checking changes with now having `move_factors_into_sums` move some of the terms around. I am fine to release JuMP with this failure and the next release of InfiniteOpt will fix it and have a minimum compat for JuMP `v1.15`.

**Comment by odow** [comment]
@ccoffrin has asked that we hold off another week to give the LANL folks more time to test. My tentative plan is to merge and tag this on the 15th (NZ time).

File: docs/src/changelog.md [hunk 4]
```diff
@@ -7,6 +7,75 @@ CurrentModule = JuMP
...
+ - Fixed uses of `@nospecialize` which cause precompilation failures in Julia
+   v1.6.0 and v1.6.1. (#3464)
+ - Fixed adding a container of [`Parameter`](@ref) (#3473)
+ - Fixed return type of `x^0` and `x^1` to no longer return `QuadExpr` (see note
+   in `Breaking` section above) (#3474)
+
+### Other
+
+ - Added GAMS to solver documentation (#3357)
+ - Updated various tutorials (#3459) (#3460) (#3462) (#3463) (#3465)
```
Comment (odow) [inline docs/src/changelog.md:74]:
```suggestion
 - Updated various tutorials (#3459) (#3460) (#3462) (#3463) (#3465) (#3490) (#3492)
 - Added [The network multi-commodity flow problem](@ref) tutorial (#3491)
```

File: docs/src/changelog.md [hunk 5]
```diff
@@ -7,6 +7,75 @@ CurrentModule = JuMP
...
+   type-stable `x * x` instead of `x^2`.
+
+### Added
+
+ - Added [`triangle_vec`](@ref) which simplifies adding [`MOI.LogDetConeTriangle`](@ref)
+   and [`MOI.RootDetConeTriangle`](@ref) constraints (#3456)
+ - Added the new nonlinear interface. This is a very large change. See the
+   documentation at [Nonlinear Modeling](@ref) and the (long) discussion in
+   [JuMP.jl#3106](https://github.com/jump-dev/JuMP.jl/pull/3106). Related PRs
+   are (#3468) (#3472) (#3475) (#3483) (#3487) (#3488)
```
Comment (odow) [inline docs/src/changelog.md:61]:
```suggestion
   are (#3468) (#3472) (#3475) (#3483) (#3487) (#3488) (#3489)
```

**Review by mlubin — APPROVED** [review]
Change log looks good

**Comment by odow** [comment]
Looks like there's still a failure in UnitJuMP. It might actually be our problem.

**Comment by odow** [comment]
Latest release of UnitJuMP is now passing. Thanks @trulsf

**Comment by odow** [comment]
#3500 is not blocking (it's also a problem in 1.14.1), so I plan to merge this tomorrow during my call with @ccoffrin (09:00 NZ time).

File: docs/src/changelog.md [hunk 6]
```diff
@@ -7,6 +7,82 @@ CurrentModule = JuMP
...
+   type-stable `x * x` instead of `x^2`.
+
+### Added
+
+ - Added [`triangle_vec`](@ref) which simplifies adding [`MOI.LogDetConeTriangle`](@ref)
+   and [`MOI.RootDetConeTriangle`](@ref) constraints (#3456)
+ - Added the new nonlinear interface. This is a very large change. See the
+   documentation at [Nonlinear Modeling](@ref) and the (long) discussion in
+   [JuMP.jl#3106](https://github.com/jump-dev/JuMP.jl/pull/3106). Related PRs
+   are (#3468) (#3472) (#3475) (#3483) (#3487) (#3488) (#3489) (#3504)
```
Comment (odow) [inline docs/src/changelog.md:61]:
```suggestion
   are (#3468) (#3472) (#3475) (#3483) (#3487) (#3488) (#3489) (#3504) (#3509)
```

**Comment by odow** [comment]
Just running https://github.com/jump-dev/JuMP.jl/actions/runs/6191628679 one last time

**Pull request merged**